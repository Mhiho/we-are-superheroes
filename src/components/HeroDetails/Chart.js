import React from "react";
import "../../index.css";

const Chart = (props) => {
  return (
    <div className="chart-container d-flex justify-content-center">
      <div className="chart animation">
        <p className="text-center">{props.percentage}</p>
        <div
          style={{
            background: "#363636",
            width: "30px",
            margin: "auto",
            height: `${100 - props.percentage}px`,
          }}
          className="resize"
        ></div>
        <div
          style={{
            background: "gold",
            width: "30px",
            height: `${props.percentage}px`,
            margin: "auto",
          }}
          className="chart-column"
        ></div>
        <h6 className="text-center">{props.label}</h6>
      </div>
    </div>
  );
};
export default Chart;
