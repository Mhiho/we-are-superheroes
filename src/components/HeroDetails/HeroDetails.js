import React from 'react'
import '../../index.css'

const heroDetails = props => {
    return (
        <div  className="fixed-position">
            <span style={{color: 'white'}} onClick={props.close}>x</span>
            <h1>{props.name}</h1>
            <img src={props.src} alt="imageBigHero"/>
            <h3>bio: </h3>
            <h3>
                powers:
            </h3>
        </div>
    )
}
export default heroDetails;