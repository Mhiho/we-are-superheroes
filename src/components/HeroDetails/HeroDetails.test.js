import React from 'react';
import renderer from 'react-test-renderer';

import HeroDetails from './HeroDetails';

it('renders correctly when <HeroDetails /> is null', () => {
  const tree = renderer.create(null).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly where there is object as props', ()=>{

        let values1 = ['one','two']
        let keys1 = ['one','two']
        let values2 = ['one','two']
        let keys2 = ['one','two']
        const tree = renderer.create(<HeroDetails biokeys={keys1} bioels={values1} statkeys={keys2} statels={values2} />).toJSON();
        expect(tree).toMatchSnapshot();
})
