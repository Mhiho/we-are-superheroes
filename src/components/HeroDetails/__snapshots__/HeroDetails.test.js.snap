// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when <HeroDetails /> is null 1`] = `null`;

exports[`renders correctly where there is object as props 1`] = `
<div
  className="jumbotron fixed-position w-100 media-query-jumbotron"
  style={
    Object {
      "color": "white",
    }
  }
>
  <h1
    className="positioned-x"
  >
    <span
      className="float-right"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      âœ˜
    </span>
  </h1>
  <p
    className="lead"
  >
    <div
      className="d-flex align-items-start justify-content-start flex-col media-query-class"
    >
      <img
        alt="imageBigHero"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="p-2 container-hero-details"
      >
        <h1
          className="text-center text-danger"
        />
        <div
          className="d-flex flex-row"
        >
          <h3
            className="d-flex align-items-center text-warning"
          >
            Powers: 
          </h3>
          <div
            className="d-flex flex-col align-items-start additional-class"
          >
            <div
              className="chart-container d-flex justify-content-center"
            >
              <div
                className="chart animation"
              >
                <p
                  className="text-center"
                >
                  one
                </p>
                <div
                  className="resize"
                  style={
                    Object {
                      "background": "#363636",
                      "height": "NaNpx",
                      "margin": "auto",
                      "width": "30px",
                    }
                  }
                />
                <div
                  className="chart-column"
                  style={
                    Object {
                      "background": "gold",
                      "height": "onepx",
                      "margin": "auto",
                      "width": "30px",
                    }
                  }
                />
                <h6
                  className="text-center"
                >
                  0
                </h6>
              </div>
            </div>
          </div>
          <div
            className="d-flex flex-col align-items-start additional-class"
          >
            <div
              className="chart-container d-flex justify-content-center"
            >
              <div
                className="chart animation"
              >
                <p
                  className="text-center"
                >
                  two
                </p>
                <div
                  className="resize"
                  style={
                    Object {
                      "background": "#363636",
                      "height": "NaNpx",
                      "margin": "auto",
                      "width": "30px",
                    }
                  }
                />
                <div
                  className="chart-column"
                  style={
                    Object {
                      "background": "gold",
                      "height": "twopx",
                      "margin": "auto",
                      "width": "30px",
                    }
                  }
                />
                <h6
                  className="text-center"
                >
                  1
                </h6>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h3
            className="text-warning"
          >
            Biography: 
          </h3>
          <h4>
            <span>
              0
              : 
            </span>
            one
          </h4>
          <h4>
            <span>
              1
              : 
            </span>
            two
          </h4>
        </div>
      </div>
    </div>
  </p>
</div>
`;
